app/services/gatherers/vote_account_details_service.rb

```diff
@@ -9,25 +9,39 @@ class VoteAccountDetailsService
     def initialize(network:, config_urls:)
       @network = network
       @config_urls = config_urls
+      @range_start = VoteAccount.where(network: @network).order(id: :asc).first.id
+      @range_end = VoteAccount.where(network: @network).order(id: :asc).last.id
     end
 
     def call
-      VoteAccount.where(network: @network).each do |vacc|
+      VoteAccount.where(id: @range_start..@range_end, network: @network)
+                 .order(id: :asc)
+                 .each do |vacc|
+        @range_start = vacc.id
+
         vote_account_details = get_vote_account_details(vacc.account)
-        unless vote_account_details.blank?
-          vacc.update(
+        if vote_account_details.blank?
+          vacc.set_inactive
+        else
+          if vacc.update(
```

Comment:
You should check here if the values from solana are different than what we already have nad based on that update or not the record